### PR TITLE
Goodbye fqname

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Area of change
- Bring back FQTypeName, FQFnName, and FQConstantName, and remove FQName
- Remove Built-in types
```

### CODE REVIEW

File: backend/src/BuiltinExecution/Libs/Json.fs [hunk 1]
```diff
@@ -205,7 +205,7 @@ let rec serialize
               cases
               |> NEList.find (fun c -> c.name = caseName)
               |> Exception.unwrapOptionInternal
-                "Couldn't find matching case"
+                $"Couldn't find matching case for {caseName}"
```
Comment (OceanOak) [inline backend/src/BuiltinExecution/Libs/Json.fs:208]: Todo: revert this change it was just used for some debugging

File: backend/src/LibBinarySerialization/ProgramTypesToSerializedTypes.fs [hunk 2]
```diff
@@ -181,54 +93,232 @@ module NameResolution =
     | Error err -> Error(NameResolutionError.Error.toPT err)
 
 
+(*
```
Comment (OceanOak) [inline backend/src/LibBinarySerialization/ProgramTypesToSerializedTypes.fs:96]: TODO: check the commented out code

File: backend/src/LibCliExecution/PackageManager.fs [hunk 3]
```diff
@@ -804,26 +772,26 @@ let packageManager (baseUrl : string) : RT.PackageManager =
...
           name.owner
           name.modules
-          typeName
+          name.name
           name.version
           conversionFn)
 
     getFn =
-      withCache (fun ({ name = RT.FnName.FnName fnName } as name) ->
+      withCache (fun (name) ->
```
Comment (OceanOak) [inline backend/src/LibCliExecution/PackageManager.fs:787]: TODO remove parans

File: backend/src/LibCliExecution/PackageManager.fs [hunk 4]
```diff
@@ -834,14 +802,14 @@ let packageManager (baseUrl : string) : RT.PackageManager =
         fetchById "function" tlid conversionFn)
 
     getConstant =
-      withCache (fun ({ name = RT.ConstantName.ConstantName constName } as name) ->
+      withCache (fun (name) ->
```
Comment (OceanOak) [inline backend/src/LibCliExecution/PackageManager.fs:805]: TODO: remove parans

File: backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs [hunk 5]
```diff
@@ -7,52 +7,116 @@ open Prelude
...
+    match fqtn with
+    | PT.FQTypeName.Package p -> RT.FQTypeName.Package(Package.toRT p)
+    | PT.FQTypeName.UserProgram u -> RT.FQTypeName.UserProgram(UserProgram.toRT u)
+
+  let fromRT (fqtn : RT.FQTypeName.FQTypeName) : Option<PT.FQTypeName.FQTypeName> =
+    match fqtn with
+    | RT.FQTypeName.Package p -> PT.FQTypeName.Package(Package.fromRT p) |> Some
+    | RT.FQTypeName.UserProgram u -> PT.FQTypeName.UserProgram(UserProgram.fromRT u) |> Some
+
+
```
Comment (OceanOak) [inline backend/src/LibExecution/ProgramTypesToRuntimeTypes.fs:41]: TODO: swap const and fn order

File: backend/src/LibExecution/RuntimeTypes.fs [hunk 6]
```diff
@@ -32,304 +32,287 @@ open FSharp.Control.Tasks
...
+  let builtin (modules : List<string>) (name : string) (version : int) : Builtin =
+    assert' modules name version assertFnName
+    { modules = modules; name = name; version = version }
 
-  let fqBuiltIn (modules : List<string>) (name : string) (version : int) : FnName =
-    FQName.fqBuiltIn assert' modules (FnName name) version
+  let fqBuiltin (modules : List<string>) (name : string) (version : int) : FQFnName =
+    Builtin(builtin modules name version)
 
   let userProgram
```
Comment (OceanOak) [inline backend/src/LibExecution/RuntimeTypes.fs:166]: TODO: swap package and user fns order

File: packages/darklang/languageTools/programTypes.dark [hunk 7]
```diff
@@ -1,60 +1,54 @@
...
-      /// Used to name where type/function/etc lives, eg a BuiltIn module, a User module,
-      /// or a Package module.
-      module FQName =
 
-        /// A name that is built into the runtime
-        type BuiltIn<'name> =
-          { modules: List<String>
-            name: 'name
-            version: Int64 }
+      module FQTypeName =
```
Comment (OceanOak) [inline packages/darklang/languageTools/programTypes.dark:5]: Todo add back the comment 

/// Used to name where type/function/etc lives, eg a BuiltIn module, a User module,
      /// or a Package module.

File: packages/darklang/languageTools/programTypes.dark [hunk 8]
```diff
@@ -1,60 +1,54 @@
...
 
-        /// Part of the user's program (eg canvas or cli)
-        type UserProgram<'name> =
-          { modules: List<String>
-            name: 'name
-            version: Int64 }
+        type UserProgram = { modules: List<String>; name: String; version: Int64 }
```
Comment (OceanOak) [inline packages/darklang/languageTools/programTypes.dark:7]: TODO: add comment back
/// Part of the user's program (eg canvas or cli)
type UserProgram

File: packages/darklang/languageTools/programTypes.dark [hunk 9]
```diff
@@ -1,60 +1,54 @@
...
 
-        /// The name of a thing in the package manager
-        // TODO: We plan to use UUIDs for this, but this is a placeholder
-        type Package<'name> =
-          { owner: String
-            modules: List<String>
-            name: 'name
-            version: Int64 }
+        type Package =
+          { owner : String
```
Comment (OceanOak) [inline packages/darklang/languageTools/programTypes.dark:10]: Todo: add comment and todo 
/// The name of a thing in the package manager
        // TODO: We plan to use UUIDs for this, but this is a placeholder

File: packages/darklang/languageTools/runtimeTypes.dark [hunk 10]
```diff
@@ -5,57 +5,55 @@ module Darklang =
...
-
-      module FnName =
-        type Name = FnName of String
-        type FnName = FQName.FQName<Name>
-        type BuiltIn = FQName.BuiltIn<Name>
-        type UserProgram = FQName.UserProgram<Name>
-        type Package = FQName.Package<Name>
+      module FQTypeName =
+
+        type UserProgram = { modules: List<String>; name: String; version: Int64 }
```
Comment (OceanOak) [inline packages/darklang/languageTools/runtimeTypes.dark:10]: TODO: add comment 
/// Part of the user's program (e.g. canvas or CLI)

File: packages/darklang/languageTools/semanticTokens.dark [hunk 11]
```diff
@@ -98,9 +98,9 @@ module Darklang =
 
 
       module TypeReference =
-        module BuiltIn =
+        module Builtin =
```
Comment (OceanOak) [inline packages/darklang/languageTools/semanticTokens.dark:101]: Todo: check if this was meant to be renamed from BuiltIn to Builtin

File: packages/darklang/languageTools/writtenTypes.dark [hunk 12]
```diff
@@ -34,7 +34,7 @@ module Darklang =
 
       // Types
       module TypeReference =
-        type BuiltIn =
+        type Builtin =
```
Comment (OceanOak) [inline packages/darklang/languageTools/writtenTypes.dark:37]: Todo: check if this was meant to be renamed from BuiltIn to Builtin

File: packages/darklang/languageTools/writtenTypesToProgramTypes.dark [hunk 13]
```diff
@@ -57,9 +57,9 @@ module Darklang =
 
 
       module TypeReference =
-        module BuiltIn =
+        module Builtin =
```
Comment (OceanOak) [inline packages/darklang/languageTools/writtenTypesToProgramTypes.dark:60]: Todo: check if this was meant to be renamed from BuiltIn to Builtin

File: packages/darklang/prettyPrinter/programTypes.dark [hunk 14]
```diff
@@ -102,223 +83,271 @@ module Darklang =
...
+
+            let versionPart =
+              if u.version == 0L then
+                ""
+              else
+                $"_v{PACKAGE.Darklang.Stdlib.Int64.toString u.version}"
+
+            $"{modulesPart}{namePart}{versionPart}"
 
         module Package =
```
Comment (OceanOak) [inline packages/darklang/prettyPrinter/programTypes.dark:295]: TODO: swap package and userProgram order?

File: backend/src/LibParser/NameResolver.fs [hunk 15]
```diff
@@ -347,8 +576,8 @@ let resolve
 module TypeName =
```
Comment (OceanOak) [inline backend/src/LibParser/NameResolver.fs:576]: Is this supposed to be renamed to FQTypeName?

File: packages/darklang/languageTools/writtenTypes.dark (hunk 12 above)
Comment (StachuDotNet) [reply packages/darklang/languageTools/writtenTypes.dark:37]: Yeah - at least, at this point I believe the majority of the usages are typed this way, so let's be consistent like this.

File: backend/src/BuiltinCliHost/Libs/Cli.fs [hunk 16]
```diff
@@ -19,6 +19,11 @@ module WT = LibParser.WrittenTypes
 module Json = BuiltinExecution.Libs.Json
 
 
+let typ =
+  FQTypeName.Package
+    { owner = "Darklang"; modules = [ "Cli" ]; name = "ExecutionError"; version = 0 }
```
Comment (StachuDotNet) [inline backend/src/BuiltinCliHost/Libs/Cli.fs:24]: love that this is extracted, but let's rename to `executionErrorTyp` or something like that

Comment (StachuDotNet) [reply backend/src/BuiltinCliHost/Libs/Cli.fs:24]: or maybe something like

```fsharp
module ExecutionError = 
  let fqTypeName = FQTypeName....
  let typeRef = ...
```

since many/most uses are type refs
